### DISCUSSION TITLE
Welcome to CameraScanner.Maui Discussions!

### DISCUSSION THREAD

**Discussion by thomasgalliker** [thomasgalliker/CameraScanner.Maui#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by deltaly23** [comment]
thomasgalliker, please help me, i cant change between front/back camera on android :(

**Reply by thomasgalliker** [reply]
Hi @deltaly23! Did you have a look at the sample app provided in this repository? There is a property "CameraFacing" which can be "Front" or "Back". You'll find a sample usage in [CameraPreviewPage.xaml](https://github.com/thomasgalliker/CameraScanner.Maui/blob/1e2ecf8808247c0d58ba601aa5855bb204686535/Samples/CameraDemoApp/Views/CameraPreviewPage.xaml#L27).

**Reply by deltaly23** [reply]
hi @thomasgalliker, it doesnt work even on the sample app CamerapreviewPage, heres in the ateachment of the log if you can help me.
[app_log.txt](https://github.com/user-attachments/files/19549957/app_log.txt)

**Reply by thomasgalliker** [reply]
Strange. What device are you using?

**Reply by deltaly23** [reply]
i try on android emulator pixel 7, but, while reading and debugging the sample app and i found a condition that was blocking the execution of .UpdateCamera on CameraManager:
![image](https://github.com/user-attachments/assets/8a61a8fe-d5bf-4dbc-836b-456fece343ef)

**Reply by thomasgalliker** [reply]
I just tried on a Pixel 7 API 35 emulator. It works. I'll have a look at the code you posted later. What do you think is wrong there?
Did you request the camera permissions? If not, the screen remains black on the simulator (and on devices too, of course).

**Reply by deltaly23** [reply]
yeah, the request permission are setted. For some reason, This.cameraController.CameraSelector arent null when i update the camera facing, because of that the condition are blocking the update.
I remove the condition in the CameraManager.StartAsync and it works, you think that could cause some collateral effects in the code?

**Reply by thomasgalliker** [reply]
Strange. I'll have a look at this line. The thing is, it's pretty difficult to coordinate all "update calls" from properties in order to have an efficient and race condition free startup of the camera. Thanks for the hint. I wonder what the difference of our android emulators is? Can you take a screenshot of the configuration somehow?

**Reply by thomasgalliker** [reply]
There is a nuget pre-release available for you: CameraScanner.Maui 1.5.1-pre contains optimizations around the line of code you mentioned. There was a potential problem (but not debuggable for myself) for which I guess I found a solution. Give it a try and let me know if it helps.

**Reply by thomasgalliker** [reply]
Let's move the discussion to this issue: https://github.com/thomasgalliker/CameraScanner.Maui/issues/49
@deltaly23 your feedback to the CameraScanner.Maui 1.5.1-pre? Does it work with this release?

**Comment by l0gaw** [comment]
How can I create PR to this Repo?

**Reply by thomasgalliker** [reply]
Fork the repository and create a pull request from your forked branch to this repository's develop branch. More infos [here](https://docs.github.com/de/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork).

If you want to contribute, I'd start with a [new issue](https://github.com/thomasgalliker/CameraScanner.Maui/issues/new/choose) where your explain what you'd like to change or where you see a bug.

**Comment by SandroK95** [comment]
Hello Thomas! 
I have a question regarding the barcode scanning capabilities. Is it possible to scan a DataMatrix code that doesn't have any borders or quiet zone around it? Thanks a lot!

**Reply by thomasgalliker** [reply]
Hi. There is a sample app in this repository which you can use to try. DataMatrix is definitely supported by both, iOS and Android native barcode scanning capabilities. If you have a specific DataMatrix barcode which you want to test, you can attach it here and I can try.

**Reply by SandroK95** [reply]
Hi Thomas, thanks a lot for answer. I really appreciate it
I will give you one work document with few examples, you will see that first datamatrix is not able to scan, but second one i can because it has bounderies.
[data matrix_examples.docx](https://github.com/user-attachments/files/21210434/data.matrix_examples.docx)

**Reply by thomasgalliker** [reply]
Yes, same here: The first DataMatrix cannot be scanned, the following three can be scanned. I'm not an expert for DM but I guess the border should be part of the barcode.

**Reply by SandroK95** [reply]
Probably, I will research more ofc :) Thanks again for answer!

**Reply by SandroK95** [reply]
Thomas sorry for bothering you, one more question. Is there any option to mark Barcode that i scanned? If i need to scan 5 barcodes for example. Thanks in advance!

**Reply by thomasgalliker** [reply]
I don't understand the question. You want to scan 5 different barcodes and get to know whether one of them was already scanned?

**Reply by SandroK95** [reply]
![Screenshot_20250715_150934](https://github.com/user-attachments/assets/a6afec6e-ec24-46f4-b4ea-c692acbd9f35)
I create one application in .net Maui, and as you can see first three barcodes are marked.. (I mark it in Paint just for example) is there any option to mark it through your sdk? I hope you understand me now

**Reply by thomasgalliker** [reply]
Ok got it. Have a look at the [BarcodeResultOverlay](https://github.com/thomasgalliker/CameraScanner.Maui/blob/f4744e4677b10eee221dd862ae092c194e9bc743/Samples/CameraDemoApp/Views/UniversalScannerPage.xaml#L35) in UniversalScannerPage.xaml in the sample app which comes with this repository. There I attempt to draw colored rectangles at the position where a barcode is detected. I hope this is what you're looking for.

**Reply by SandroK95** [reply]
Thanks Thomas. I think i did it!

**Reply by thomasgalliker** [reply]
Yeah cool. Good luck with the project!
[![Buy Me a Coffee](https://img.shields.io/badge/support-buy%20me%20a%20coffee-FFDD00)](https://buymeacoffee.com/thomasgalliker)

**Comment by ibanezjp** [comment]
Hello @thomasgalliker, my name is Juan Pablo, and I'm currently using your plugin. First of all, thank you! Do you have any ideas on how to implement a "SoundOnDetected" feature, similar to VibrationOnDetected? So far, we've been using https://github.com/jfversluis/Plugin.Maui.Audio after detection, but it would be nice to have something built into your project. I'm happy to help with the coding, but I wanted to know if you already have any thoughts.

**Reply by thomasgalliker** [reply]
Sounds nice 😉 Can I ask you, why you want to eliminate the Plugin.Maui.Audio dependeny? Is the too much latency between barcode detection and sound playback?
What sound formats do you typically use? mp3, wav?

**Reply by ibanezjp** [reply]
.ogg for Android
.aiff for iOS

I don't mind to have the Plugin.Maui.Audio as a dependency if it is the only way to play a sound, but having to add it after adds complexity to the code and also is a different pattern that other team members need to learn.

**Reply by thomasgalliker** [reply]
Have a look at the [1.7.19-pre](https://www.nuget.org/packages/CameraScanner.Maui/1.7.19-pre) release of CameraScanner.Maui. CameraView contains new properties SoundOnDetected, SoundSource and SoundVolume. Let's move the discussion to issue #66 👍🏻
